dot/core/service.go

```diff
@@ -190,12 +198,12 @@ func (s *Service) HandleBlockProduced(block *types.Block, state *rtstorage.TrieS
 }
 
 func (s *Service) handleBlock(block *types.Block, state *rtstorage.TrieState) error {
-	if block == nil || block.Header == nil || state == nil {
+	if block == nil || state == nil {
```

Comment:
shouldn't this return an error?  Why would the block and state be `nil`? @noot

Reply:
yeah, it shouldn't be nil, should return an error

Reply:
Sorry I'm a little confused how to act upon this. Are you just saying that this nil check be returning an error instead of nil? Also the new header type cannot be compared to nil like it was before. I figured the check was there since the old block header was a pointer, and now that its not I removed that check. However if there is more to this check than I thought, I would need some insight as to how to approach that @timwu20 @noot

Reply:
the former.  
> Are you just saying that this nil check be returning an error instead of nil?
this should return an error